uni-stark/tests/fib_air.rs

```diff
@@ -0,0 +1,198 @@
+use std::borrow::Borrow;
+
+use p3_air::{Air, AirBuilder, AirBuilderWithPublicValues, BaseAir};
+use p3_field::PrimeField64;
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::MatrixRowSlices;
+
+/// For testing the public values feature
+
+pub struct FibonacciAir {}
+
+impl<F> BaseAir<F> for FibonacciAir {
+    fn width(&self) -> usize {
+        NUM_FIBONACCI_COLS
+    }
+}
+
+impl<AB: AirBuilderWithPublicValues> Air<AB> for FibonacciAir {
+    fn eval(&self, builder: &mut AB) {
+        let main = builder.main();
+        let pis = builder.public_values();
+
+        let a = pis[0];
+        let b = pis[1];
+        let x = pis[2];
+
+        let local: &FibonacciRow<AB::Var> = main.row_slice(0).borrow();
+        let next: &FibonacciRow<AB::Var> = main.row_slice(1).borrow();
+
+        let mut when_first_row = builder.when_first_row();
+
+        when_first_row.assert_eq(local.left, a);
+        when_first_row.assert_eq(local.right, b);
+
+        let mut when_transition = builder.when_transition();
+
+        // a' <- b
+        when_transition.assert_eq(local.right, next.left);
+
+        // b' <- a + b
+        when_transition.assert_eq(local.left + local.right, next.right);
+
+        builder.when_last_row().assert_eq(local.right, x);
+    }
+}
+
+pub fn generate_trace_rows<F: PrimeField64>(a: u64, b: u64, n: usize) -> RowMajorMatrix<F> {
+    assert!(n.is_power_of_two());
+
+    let mut trace =
+        RowMajorMatrix::new(vec![F::zero(); n * NUM_FIBONACCI_COLS], NUM_FIBONACCI_COLS);
+
+    let (prefix, rows, suffix) = unsafe { trace.values.align_to_mut::<FibonacciRow<F>>() };
+    assert!(prefix.is_empty(), "Alignment should match");
+    assert!(suffix.is_empty(), "Alignment should match");
+    assert_eq!(rows.len(), n);
+
+    rows[0] = FibonacciRow::new(F::from_canonical_u64(a), F::from_canonical_u64(b));
+
+    for i in 1..n {
+        rows[i].left = rows[i - 1].right;
+        rows[i].right = rows[i - 1].left + rows[i - 1].right;
+    }
+
+    trace
+}
+
+const NUM_FIBONACCI_COLS: usize = 2;
+
+pub struct FibonacciRow<F> {
+    pub left: F,
+    pub right: F,
+}
+
+impl<F> FibonacciRow<F> {
+    fn new(left: F, right: F) -> FibonacciRow<F> {
+        FibonacciRow { left, right }
+    }
+}
+
+impl<F> Borrow<FibonacciRow<F>> for [F] {
+    fn borrow(&self) -> &FibonacciRow<F> {
+        debug_assert_eq!(self.len(), NUM_FIBONACCI_COLS);
+        let (prefix, shorts, suffix) = unsafe { self.align_to::<FibonacciRow<F>>() };
+        debug_assert!(prefix.is_empty(), "Alignment should match");
+        debug_assert!(suffix.is_empty(), "Alignment should match");
+        debug_assert_eq!(shorts.len(), 1);
+        &shorts[0]
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
nit: I don't think this is needed since this is an integration test? https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html

Reply:
fixed, thanks for the pointer